app/grandchallenge/reader_studies/static/reader_studies/js/display_set_create.js

```diff
@@ -0,0 +1,64 @@
+$(document).ready(() => {
+   $('#form-new-ds').on('submit', (e) => {
+      e.preventDefault();
+      const target = $(e.currentTarget);
+      $(".is-invalid").removeClass("is-invalid");
+      $(".invalid-feedback").remove();
+      $("#form-error-message").remove();
+      const formData = {};
+      const interfaces = [];
+      $.each($(target).serializeArray(), (i, entry) =>{
+          formData[entry.name] = entry.value
+      });
+      $(".extra-interface-form").each(
+          (i, interfaceForm) => {
+              const data = {}
+              $.each($(interfaceForm).serializeArray(), (i, entry) =>{
+                  data[entry.name] = entry.value
+              });
+              interfaces.push(data)
+          }
+      )
+      formData.new_interfaces = interfaces;
+      $.ajax({
+          type: 'POST',
+          url:  target.attr("action"),
+          data: JSON.stringify(formData),
```

Comment:
I guess you deliberately decided on doing this in js, but you could also add extra variables to the request with the [`hx-vars`](https://htmx.org/attributes/hx-vars/) attribute in the html (which can call a function that computes the values). Since you would still need custom js to compute the values, I'm not sure if it's worth changing, just wanted to mention it since we're trying to move away from custom js code.

Reply:
Yes we still need js, as we need to commit multiple forms at the same time basically. The form that has all the interfaces that are already in the reader study, and a form for each of the new interfaces added here. I reworked how to do that multiple times before ending up with this. I went this route because I couldn't think of a way to properly handle everything in the backend without losing information in case the form does not validate.

Reply:
Yes that makes sense. The error handling complicates things.